wordpress-mobile/WordPress-iOS
# Jetpack app: new target + duplicated app extensions

## Description
- Same idea as https://github.com/wordpress-mobile/WordPress-iOS/pull/16095, but duplicated app extensions for the Jetpack target
    - Included: TodayWidget, ThisWeekWidget, AllTimeWidget, StatsWidgets, NotificationContentExtension, NotificationServiceExtension, IntentsExtension
    - Not included: ShareExtension, DraftExtension (since the Jetpack app isn't about publishing posts)

## How to test
1. Run `bundle exec pod install`
2. Build the WordPress scheme
3. ✅ Should build as usual
3. Build the Jetpack scheme
4. ✅ Should build and it should have a 1) different icon 2) different splash screen
5. Go to My Site > Switch Site
6. ✅ A "Welcome to the Jetpack app!" notice should be displayed

## Pros / Cons

### Pros
- Easy to keep WP and Jetpack in sync
- There is no issue with bundle identifiers being changed on build time and having them polluting your git changes (like it happens with the solution in #16095)

### Cons
- If we want to change a screen that already exists in WordPress, we need to add preprocessor macros (#if JETPACK) in the original code. Either that or duplicating the file (which might become very complex and costly very quickly)
- Need to make sure that files added to the WordPress target are also added to the Jetpack target as needed (and same goes for the WordPress app extension targets)
